libs/seiscomp/io/recordstream/file.cpp

```diff
@@ -211,13 +221,44 @@ bool File::setRecordType(const char *type) {
 
 
 
+// >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
+const File::TimeWindowFilter* File::findTimeWindowFilter(Record *rec) {
+
+	// First look for fully qualified stream id (no wildcards)
+	const auto & it = _filter.find(rec->streamID());
+	if ( it != _filter.end() ) {
+		return &it->second;
+	}
+
+	// then search the wildcarded filters
+	for ( const auto& pair : _reFilter ) {
+		const string& wild = pair.first;
+		const TimeWindowFilter& twf = pair.second;
+		if ( Core::wildcmp(wild, rec->streamID()) ) {
```

Comment:
`rec->streamID()` is rather expensive as it composes the string on demand. Please save this value at the top of the function. It should be called only once.